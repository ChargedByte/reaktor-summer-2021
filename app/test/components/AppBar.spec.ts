import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils'

import Vuetify from 'vuetify'

import AppBar from '~/components/AppBar.vue'

describe('AppBar', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(AppBar, {
      localVue,
      vuetify,
      stubs: { RouterLink: RouterLinkStub },
    })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(AppBar, {
      localVue,
      vuetify,
      stubs: { RouterLink: RouterLinkStub },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
