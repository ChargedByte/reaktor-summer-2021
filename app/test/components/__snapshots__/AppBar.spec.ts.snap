// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders correctly 1`] = `
"<header class=\\"v-sheet theme--dark v-toolbar v-toolbar--dense v-app-bar v-app-bar--fixed primary\\" style=\\"height: 48px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;\\">
  <div class=\\"v-toolbar__content\\" style=\\"height: 48px;\\">
    <div class=\\"v-tabs v-tabs--centered theme--dark\\">
      <div role=\\"tablist\\" class=\\"v-item-group theme--dark v-slide-group v-tabs-bar v-tabs-bar--is-mobile white--text\\">
        <div class=\\"v-slide-group__prev v-slide-group__prev--disabled\\">
          <transition-stub name=\\"fade-transition\\"></transition-stub>
        </div>
        <div class=\\"v-slide-group__wrapper\\">
          <div class=\\"v-slide-group__content v-tabs-bar__content\\">
            <div class=\\"v-tabs-slider-wrapper\\" style=\\"transition: none;\\">
              <div class=\\"v-tabs-slider\\"></div>
            </div><a tabindex=\\"0\\" aria-selected=\\"true\\" role=\\"tab\\" class=\\"v-tab\\">
              beanies
            </a><a tabindex=\\"0\\" aria-selected=\\"false\\" role=\\"tab\\" class=\\"v-tab\\">
              facemasks
            </a><a tabindex=\\"0\\" aria-selected=\\"false\\" role=\\"tab\\" class=\\"v-tab\\">
              gloves
            </a>
          </div>
        </div>
        <div class=\\"v-slide-group__next v-slide-group__next--disabled\\">
          <transition-stub name=\\"fade-transition\\"></transition-stub>
        </div>
      </div>
    </div>
  </div>
</header>"
`;
